Sephora-M/assignment1/SinglyLinkedListTest.java

```diff
@@ -0,0 +1,49 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.ExpectedException;
+
+class SinglyLinkedListTest {
+
+	
+	@Rule
+    public ExpectedException thrown = ExpectedException.none();
+	
+	@Test
+	void testKToLast() {
+		SinglyLinkedListNode<Integer> one = new SinglyLinkedListNode<Integer>(1);
+		SinglyLinkedListNode<Integer> two = new SinglyLinkedListNode<Integer>(2);
+		SinglyLinkedListNode<Integer> three = new SinglyLinkedListNode<Integer>(3);
+		SinglyLinkedListNode<Integer> four = new SinglyLinkedListNode<Integer>(4);
+		SinglyLinkedListNode<Integer> five = new SinglyLinkedListNode<Integer>(5);
+		SinglyLinkedListNode<Integer> six = new SinglyLinkedListNode<Integer>(6);
+		SinglyLinkedListNode<Integer> seven = new SinglyLinkedListNode<Integer>(7);
+		SinglyLinkedListNode<Integer> eight = new SinglyLinkedListNode<Integer>(8);
+		SinglyLinkedListNode<Integer> nine = new SinglyLinkedListNode<Integer>(9);
+		SinglyLinkedListNode<Integer> ten = new SinglyLinkedListNode<Integer>(10);
+		
+		one.setNext(two);
+		two.setNext(three);
+		three.setNext(four);
+		four.setNext(five);
+		five.setNext(six);
+		six.setNext(seven);
+		seven.setNext(eight);
+		eight.setNext(nine);
+		nine.setNext(ten);
+		
+		assertEquals(one.kToLast(2), eight);
```

Comment:
Instead checking object instances, you can do kToLast(2).getData() and check just the integer value.

Reply:
I did that initially, but what if we the kToLast method returns a node that is not the kth to last but happens to have the same value stored as the kth to last element?